### DISCUSSION TITLE
Make useQuery() return isRetrying

### DISCUSSION THREAD

**Discussion by Lesik** [TanStack/query#2211] [Ideas] [answered]
Hi,

I only recently started using React-Query, my apologies if I misunderstood something.

I felt it might be a cool feature if `useQuery()` could also return `isRetrying` in order to show some kind of "Whoops, this is taking longer than expected, hang in tight" for better UX. If your query is large and takes a while to load, but fails and you have retries set to true, the user is presented with a very long-lasting loading page with no feedback what's happening.

Example use case:

```tsx
const { isLoading, isRetrying, data } = useQuery();

if (isRetrying) {
  return <p>Whoops, this is taking longer than expected, hang in tight!</p>;
}
if (isLoading) {
  return <p>Loading...</p>;
}
```

**Comment by TkDodo** [comment, marked as answer]
I think the returned `failureCount` would help in this case.
